webui/react/src/components/WorkspaceCreateModal.tsx

```diff
@@ -42,12 +49,52 @@ const CodeEditor = React.lazy(() => import('hew/CodeEditor'));
 
 const WorkspaceCreateModalComponent: React.FC<Props> = ({ onClose, workspaceId }: Props = {}) => {
   const idPrefix = useId();
-  const { canModifyWorkspaceAgentUserGroup, canModifyWorkspaceCheckpointStorage } =
-    usePermissions();
+  const {
+    canModifyWorkspaceAgentUserGroup,
+    canModifyWorkspaceCheckpointStorage,
+    canSetWorkspaceNamespaceBindings,
+  } = usePermissions();
   const [form] = Form.useForm<FormInputs>();
   const useAgentUser = Form.useWatch('useAgentUser', form);
   const useAgentGroup = Form.useWatch('useAgentGroup', form);
   const useCheckpointStorage = Form.useWatch('useCheckpointStorage', form);
+  const resourceManagers = useAsync(async (canceller) => {
+    try {
+      const response = await getKubernetesResourceManagers(undefined, { signal: canceller.signal });
+      return response.names;
+    } catch (e) {
+      handleError(e, {
+        level: ErrorLevel.Error,
+        publicMessage: 'Failed to fetch Resource Managers.',
+        silent: false,
+        type: ErrorType.Server,
+      });
+      return [];
```

Comment:
nit: returning `Failed` when the request fails will make our lives easier when debugging issues down the line:
```suggestion
      return Failed(e);
```

Reply:
So when I try this I get this error: `Argument of type 'unknown' is not assignable to parameter of type 'Error'.`
Is it okay if I `return NotLoaded` instead?